tools/changelog/Makefile

```diff
@@ -0,0 +1,32 @@
+OS ?= $(shell uname -s | tr '[[:upper:]]' '[[:lower:]]')
+ARCH ?= $(shell uname -m)
+TOOL_NAME ?= changelog
+BUILD_DIR ?= build/$(OS)/$(ARCH)/
+VERSION ?= v0.0.0-dev
```

Comment:
It is not typical to use `?=` to assign variables as it allows environment variables to override make variables, and this can cause unusual build issues as different people may have different environments.

You can define these with `=` and they are still overridable on the `make` command line. e.g:
```
make tarball ARCH=amd64 OS=linux
```

This still overrides these default values as command line values override makefile variables.